browser/importer/brave_profile_writer.cc

```diff
@@ -63,3 +65,42 @@ void BraveProfileWriter::UpdateStats(const BraveStats& stats) {
                      https_upgrades + stats.httpsEverywhere_count);
   }
 }
+
+void BraveProfileWriter::UpdateReferral(const BraveReferral& referral) {
+  PrefService* local_state = g_browser_process->local_state();
+  if (!local_state) {
+    LOG(ERROR) << "Unable to get local_state! (needed to set referral info)";
+  }
+
+  if (!referral.week_of_installation.empty()) {
+    LOG(INFO) << "Setting kWeekOfInstallation to "
+      << "\"" << referral.week_of_installation << "\"";
+    local_state->SetString(kWeekOfInstallation, referral.week_of_installation);
+  }
+
+  if (!referral.promo_code.empty() &&
+    referral.promo_code.compare("none") != 0) {
+    LOG(INFO) << "Setting kReferralPromoCode to "
+      << "\"" << referral.promo_code << "\"";
+    local_state->SetString(kReferralPromoCode, referral.promo_code);
+  } else {
+    local_state->ClearPref(kReferralPromoCode);
```

Comment:
These prefs are also used by brave-core's `BraveReferralsService`. Is there another way to set these prefs besides importing them from Muon? It seems possible for this code to accidentally overwrite legitimate preset pref values, but I am not sure if such a scenario is actually possible or worth being concerned about.

Reply:
That's a good concern- thanks for bringing it up 😄

Reply:
Yeah, really good point. It could be useful to log the old value here, in case a user unintentionally loses data that they care about ("Setting kReferralDownloadID to FOO (was BAR)"). It might give QA a way to reset them.

Reply:
Per convo with @davidtemkin, the existing logic should be good to go 😄 👍 
https://bravesoftware.slack.com/archives/C8D9C41QQ/p1542845717143900